### PR TITLE
WorkOrders List | Add Filter

### PR DESCRIPTION
**Pull request by tiagobastos**

### CODE REVIEW

File: apiary.apib [hunk 1]
```diff
@@ -3566,6 +3566,18 @@ Get a list of work orders.
 + Request (application/json)
 
     + Attributes (object)
+        + filter (object, optional)
+            + customer (object, optional)
```
Comment (JordyDeGroeve) [inline apiary.apib]: B: don't include the filter on customer yet.

**Review by JordyDeGroeve — DISMISSED** [review]
Meant to block, sorry

File: apiary.apib (hunk 1 above)
Comment (tiagobastos) [reply apiary.apib]: 🤦‍♂

**Review by JordyDeGroeve — COMMENTED** [review]
Q: do we only allow 1 state to be passed?

**Comment by tiagobastos** [comment]
I think so yes. 
The state field is enumerating possible members.

File: apiary.apib [hunk 2]
```diff
@@ -3566,6 +3566,12 @@ Get a list of work orders.
 + Request (application/json)
 
     + Attributes (object)
+        + filter (object, optional)
+            + state (enum[string], optional)
```
Comment (WouterSioen) [inline apiary.apib]: This should be array of state, right? To allow multiple states at once?

Comment (JordyDeGroeve) [reply apiary.apib]: Let's go for the multiple states. 👍

**Review by dietrichm — DISMISSED** [review]
B: we'll be allowing filters on multiple states.

File: apiary.apib [hunk 3]
```diff
@@ -3567,11 +3567,12 @@ Get a list of work orders.
 
     + Attributes (object)
         + filter (object, optional)
-            + state (enum[string], optional)
-                + Members
-                    + draft
-                    + finalized
-                    + sent
+            + state: `draft`, `sent` (array[string], optional) - The states you want to filter by
+                + (enum[string], required)
```
Comment (dietrichm) [inline apiary.apib]: Strong PP: you can use
```
            + state (array[enum[string]], optional)
                + Members
                    + draft
                    + finalized
                    + sent
```
The example enum values tend to mess up the preview in the live docs. Also the remark is not strictly needed (pretty obvious what it does).

**Review by JordyDeGroeve — APPROVED** [review]

**Review by dietrichm — APPROVED** [review]

**Review by WouterSioen — APPROVED** [review]

**Pull request merged**